src/OrbitBase/FileTest.cpp

```diff
@@ -83,6 +83,65 @@ TEST(File, OpenNewFileForReadWrite) {
   ASSERT_TRUE(fd_or_error.has_value()) << fd_or_error.error().message();
 }
 
+TEST(File, WriteFullySmoke) {
+  auto temporary_file_or_error = TemporaryFile::Create();
+  ASSERT_TRUE(temporary_file_or_error.has_value()) << temporary_file_or_error.error().message();
+  TemporaryFile temporary_file = std::move(temporary_file_or_error.value());
+
+  // Write buffer into file.
+  const std::string buffer = "blub\nbla\n";
+  ErrorMessageOr<void> write_result_or_error = WriteFully(temporary_file.fd(), buffer);
+  ASSERT_FALSE(write_result_or_error.has_error()) << write_result_or_error.error().message();
+
+  // Read back and compare content.
+  ErrorMessageOr<unique_fd> fd_or_error = OpenFileForReading(temporary_file.file_path().string());
+  ASSERT_FALSE(fd_or_error.has_error()) << fd_or_error.error().message();
+  std::array<char, 64> read_back = {};
+  ErrorMessageOr<size_t> result_or_error = ReadFully(fd_or_error.value(), read_back.data(), 9);
```

Comment:
maybe read 11 to check that it doesn't read past of eof. also use buffer.size()+3 instead of hardcoding the size of the buffer.

Reply:
Reading back buffer.size()+3 and check the result for size buffer.size() to check that nothing got written beyond what was intended.